main.nf

```diff
@@ -511,78 +527,101 @@ if (params.skip_alignment) {
-    if (params.aligner == 'minimap2') {
+    process GTFToBED {
+        tag "$gtf"
+        label 'process_low'
 
-        process MiniMap2Index {
-          tag "$fasta"
-          label 'process_medium'
+        input:
+        set file(gtf), val(name) from ch_fastq_gtf
 
-          input:
-          set val(name), file(fasta) from ch_fasta_index
+        output:
+        set file("*.bed"), val(name) into ch_gtf_bed
 
-          output:
-          set val(name), file("*.mmi") into ch_index
-          file "*.version" into ch_minimap2_version
+        script: // This script is bundled with the pipeline, in nf-core/nanoseq/bin/
+        """
+        gtf2bed $gtf > ${gtf.baseName}.bed
+        """
+    }
 
-          script:
-          minimap_preset = (params.protocol == 'DNA') ? "-ax map-ont" : "-ax splice"
-          kmer = (params.protocol == 'directRNA') ? "-k14" : ""
-          stranded = (params.stranded || params.protocol == 'directRNA') ? "-uf" : ""
-          """
-          minimap2 $minimap_preset $kmer $stranded -t $task.cpus -d ${fasta}.mmi $fasta
-          minimap2 --version &> minimap2.version
-          """
-        }
-        ch_graphmap2_version = Channel.empty()
+    ch_chrom_sizes
+        .join(ch_gtf_bed, by: 1, remainder:true)
+        .map { it -> [ it[1], it[2], it[0] ] }
+        .cross(ch_fastq_index) { it -> it[-1] }
+        .flatten()
+        .collate(9)
+        .map { it -> [ it[5], it[0], it[6], it[1], it[7], it[8] ]} // [ fasta, sizes, gtf, bed, is_transcripts, annotation_str ]
+        .unique()
+        .set { ch_fasta_index }
 
-    } else if (params.aligner == 'graphmap2') {
+    /*
+     * STEP 8 - Create genome/transcriptome index
+     */
+    if (params.aligner == 'minimap2') {
+        process MiniMap2Index {
+            tag "$fasta"
+            label 'process_medium'
 
-        // TODO nf-core: Create graphmap2 index with GTF instead
-        // gtf = (params.protocol == 'directRNA' && params.gtf) ? "--gtf $gtf" : ""
+            input:
+            set file(fasta), file(sizes), val(gtf), val(bed), val(is_transcripts), val(annotation_str) from ch_fasta_index
+
+            output:
+            set file(fasta), file(sizes), val(gtf), val(bed), val(is_transcripts), file("*.mmi"), val(annotation_str) into ch_index
+            file "*.version" into ch_minimap2_version
+
+            script:
+            preset = (params.protocol == 'DNA' || is_transcripts) ? "-ax map-ont" : "-ax splice"
+            kmer = (params.protocol == 'directRNA') ? "-k14" : ""
+            stranded = (params.stranded || params.protocol == 'directRNA') ? "-uf" : ""
+            // TODO pipeline: Should be staging this file properly as an input
+            junctions = (params.protocol != 'DNA' && bed) ? "--junc-bed ${file(bed)}" : ""
+            """
+            minimap2 $preset $kmer $stranded $junctions -t $task.cpus -d ${fasta}.mmi $fasta
+            minimap2 --version &> minimap2.version
+            """
+        }
```

Comment:
```suggestion
        }
        ch_graphmap2_version = Channel.empty()
```

Reply:
Wheeeerrrrrreeeee are theeeesssseee? I know there is a problem but Im blind to it right now...

Reply:
Think Ive fixed this...